Makefile

```diff
@@ -1,13 +1,27 @@
 MAKEFLAGS += --silent
 GOLANGCI_LINT_VERSION = $(shell head -n 1 .golangci.yml | tr -d '\# ')
 PROTOC_VERSION := 21.12
-PROTOC_ARCHIVE := protoc-$(PROTOC_VERSION)-linux-x86_64.zip
+
+ifeq ($(OS),Windows_NT)
+    DETECTED_OS := Windows
+    PROTOC_ARCHIVE := protoc-$(PROTOC_VERSION)-win64.zip
+else
+    UNAME := $(shell uname)
+    ifeq ($(UNAME),Linux)
+        DETECTED_OS := Linux
+        PROTOC_ARCHIVE := protoc-$(PROTOC_VERSION)-linux-x86_64.zip
+    else ifeq ($(UNAME),Darwin)
+        DETECTED_OS := Darwin
+        PROTOC_ARCHIVE := protoc-$(PROTOC_VERSION)-osx-universal_binary.zip
+    endif
+endif
+
 PROTOC_DOWNLOAD_URL := https://github.com/protocolbuffers/protobuf/releases/download/v$(PROTOC_VERSION)/$(PROTOC_ARCHIVE)
 
 proto-dependencies:
 	go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.31.0
 	go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.3.0
-	@if [ "$$(uname)" != "Linux" ]; then \
+	@if [ -z "$(DETECTED_OS)" ]; then \
 		echo "Error: Can't install protoc on your OS, please install protoc-$(PROTOC_VERSION) manually." >&2; \
```

Comment:
I maintained the same logic here, but this could be improved by checking if `protoc` is already installed and skipping the download entirely. Currently, even if you have `protoc` installed manually, the task will fail if your OS doesn't match Windows, Linux, or Darwin. Should be fairly easy to add.